symbulate/results.py

```diff
@@ -330,61 +365,216 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
+                    else:
+                        ax_marg_y.hist(y, color=get_next_color(ax), normed=True,
+                                       alpha=alpha, bins=bins, orientation='horizontal')
+                plt.setp(ax_marg_x.get_xticklabels(), visible=False)
+                plt.setp(ax_marg_y.get_yticklabels(), visible=False)
+            else:
+                fig = plt.gcf()
+                ax = plt.gca()
+                color = get_next_color(ax)
+
+            nullfmt = NullFormatter() #removes labels on fig
+
+            if 'scatter' in type:
                 if jitter:
                     x += np.random.normal(loc=0, scale=.01 * (max(x) - min(x)), size=len(x))
                     y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
-                plt.scatter(x, y, color=color, alpha=alpha, **kwargs)
-            else:
-                raise Exception("I don't know how to plot these variables.")
+                ax.scatter(x, y, alpha=alpha, c=color, **kwargs)
+            elif 'hist' in type:
+                histo = ax.hist2d(x, y, bins=bins, cmap='Blues')
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=histo[3], cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
+                new_labels = []
+                for label in caxes.get_yticklabels():
+                    new_labels.append(int(label.get_text()) / len(x))
+                caxes.set_yticklabels(new_labels)
+            elif 'density' in type:
+                res = np.vstack([x, y])
+                density = gaussian_kde(res)
+                xmax, xmin = max(x), min(x)
+                ymax, ymin = max(y), min(y)
+                Xgrid, Ygrid = np.meshgrid(np.linspace(xmin, xmax, 100),
+                                           np.linspace(ymin, ymax, 100))
+                Z = density.evaluate(np.vstack([Xgrid.ravel(), Ygrid.ravel()]))
+                den = ax.imshow(Z.reshape(Xgrid.shape), origin='lower', cmap='Blues',
+                          aspect='auto', extent=[xmin, xmax, ymin, ymax]
+                )
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=den, cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
```

Comment:
Please use quotes consistently. You called it `'left'` above, but `"left"` here.